Assets/Scripts/Explodable/ExplodableObject.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ExplodableObject : MonoBehaviour
+{
+
+    private float m_TerrainRevealRadius = 30.0f;
+    private static float m_TimeOfLastSFX = 0.0f;
+
+    private int timeDlay = 3;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        Destroy(this.gameObject, timeDlay);
```

Comment:
Destroy is not called immediately, objects are scheduled for deletion and might only be destroyed a few frames in the future. The code in `OnDestroy()` may sometimes be delayed, even past the `timeDlay` setting. Calling `Destroy` over and over again every frame in `Update` may also result in unintended side effects.

You might want to use a coroutine for code delay instead:
```
private float m_ExecutionDelay = 3.0;

void Start()
{
    StartCoroutine(DoSomething);
    Destroy(gameObject, m_ExecutionDelay);
}

IEnumerator DoSomething()
{
    yield return new WaitForSeconds(m_ExecutionDelay);
    // Some code here, this will be executed after m_ExecutionDelay seconds of delay
}
```

Reply:
Wow, didn't know about that say reals!

That's a good pt to note, will update in next commit 💯